### ISSUE TITLE
Should null String field values be allowed in multipart POST requests?

### ISSUE THREAD

**Issue by piovezan** [Kong/unirest-java#398]
Not sure if this is a bug, a behavior that should be flexibilized, or one that should be enforced (or just kept that way).

**TL;DR:** Setting up a multipart POST request won't allow `field(String name, String value)` calls to receive a `null` String value as it used to in older versions (I'm updating client code dependencies and don't want to break the existing server-side code).

It throws `NullPointerException` instead since, although the [BodyPart](https://github.com/Kong/unirest-java/blob/main/unirest/src/main/java/kong/unirest/BodyPart.java) class constructor looks like it accepts `null` values (no `this.value = Objects.requireNonNull(value);` is enforced, for example), it actually throws an exception when invoking `this.partType = value.getClass();` right below in the constructor.

I'm no HTTP connoisseur but I don't think a non-`null` `String` field value should be a requirement for multipart requests (or, in the case it is, it seems like it is being only incidentally enforced). In my opinion a `null` `String` value could simply be translated to an empty string when the actual request is made.

It seems that it is only needed in order to return the `value` object's `Class<?>` instance in `getPartType()`, which in the case of a `null` value in my opinion could be the actual `T` class parameter from `BodyPart<T>` (I haven't thought out how that would translate to actual code, though). Also, I haven't cloned the repo to look for further consequences of implementing such flexibilization.

**To Reproduce**

```
import kong.unirest.Unirest;

public class Teste4
{
    public static void main(String [] args)
    {
        Unirest.post("http://www.example.com")
                .field("first", "something") // Alternatively, replace this line with .multiPartContent()
                .field("second", (String)null)
                .asJson();
    }
}
```

Resulting stack trace:

```
Exception in thread "main" java.lang.NullPointerException
	at kong.unirest.BodyPart.<init>(BodyPart.java:40)
	at kong.unirest.ParamPart.<init>(ParamPart.java:37)
	at kong.unirest.ParamPart.<init>(ParamPart.java:33)
	at kong.unirest.HttpRequestMultiPart.field(HttpRequestMultiPart.java:48)
	at Teste4.main(Teste4.java:9)
```

**Expected behavior**
In my thinking, the request object should be constructed normally instead of throwing a `NullPointerException`.

**Environmental Data**
 - Java Version: jdk-8.0.252.09-hotspot (AdoptOpenJDK, but the behavior is not tied to a specific Java version or release)
 - Version: unirest-java-3.11.10-standalone.jar (or more recent)

**Comment by piovezan** [comment]
Oh. A type erasure issue.

Not easy to retrieve the runtime type of `T`. I guess this is why `getPartType()` was needed in the first place.

Okay then, I think the issue can be closed.
